src/redux/reducer.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { AnyAction, createReducer, Draft } from '@reduxjs/toolkit';
+
+import {
+    getLocalStorageComputedLanguage,
+    getLocalStorageLanguage,
+    getLocalStorageTheme,
+    saveLocalStorageTheme,
+} from './local-storage';
+
+import {
+    ComputedLanguageAction,
+    SELECT_COMPUTED_LANGUAGE,
+    SELECT_THEME,
+    ThemeAction,
+} from './actions';
+
+import {
+    LOGOUT_ERROR,
+    RESET_AUTHENTICATION_ROUTER_ERROR,
+    SHOW_AUTH_INFO_LOGIN,
+    SIGNIN_CALLBACK_ERROR,
+    UNAUTHORIZED_USER_INFO,
+    USER,
+    USER_VALIDATION_ERROR,
+} from '@gridsuite/commons-ui';
+import { PARAM_LANGUAGE, PARAM_THEME } from '../utils/config-params';
+import { ReducerWithInitialState } from '@reduxjs/toolkit/dist/createReducer';
+
+export type AppState = {
+    computedLanguage: ReturnType<typeof getLocalStorageComputedLanguage>;
+    [PARAM_THEME]: ReturnType<typeof getLocalStorageTheme>;
+    [PARAM_LANGUAGE]: ReturnType<typeof getLocalStorageLanguage>;
+
+    user: Record<string, any> | null;
+    signInCallbackError: any;
+    authenticationRouterError: any;
+    showAuthenticationRouterLogin: boolean;
+};
```

Comment:
Actually I think that the dependency should be inverted here. 

We should define our AppState as we want it to be, and the methods that build the state should adapt to it. Here you base the type of our our entire AppState (used everywhere) on an implementation of a method. What if we want to change the method ? What if the implementation beneath changes ? We potentially have to adapt all the parts of the code that were using this state, that is not acceptable for me.
This AppState should act as an interface for our application, WE fix the interface, all the application is based on that, and the methods that feed this interface should be adapted if changes happen, not the other way around.

So it's not the type definition itself that bugs me, it's more the direction of the dependency it creates.

So for these reasons, we should avoid to defined type of AppState based on other entries, only basic types or types we created and are meaningful for us already, and absolutely avoid the any, we should define these methods as we want them in our app.